### PR TITLE
ENH prefix flags from env

### LINKED ISSUE #7: Environment dependent conda-build
Issue:

Consider an example conda recipe with the following `build.sh` script:
```
echo "LDFLAGS=$LDFLAGS"
```
When running `LDFLAGS= conda-build /path/to/example/recipe`, the output will be
```
LDFLAGS=-Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib
```
However, when `LDFLAGS` is not empty, for instance, run `LDFLAGS="-L/path/to/foo" conda-build /path/to/example/recipe`, the output is
```
LDFLAGS=-Lpath/to/foo
```

I am not sure if this is a bug or not but the result of building a conda recipe from an environment that defines `LDFLAGS` will be different from the result of the build the recipe where `LDFLAGS` is not defined. And it's not obvious to figure it out when there are some build issues because of this.

Possible options to resolve this issue: 
1. When `LDFLAGS` is defined, conda-build should prepend its flags to `LDFLAGS`.
2. When `LDFLAGS` is defined, conda-build should warn that it will use `LDFLAGS` as it is without adding its flags to `LDFLAGS`.
3. When `LDFLAGS` is defined, raise an exception.
4. When `LDFLAGS` is defined, issue a warning that `LDFLAGS` in an environment will be ignored.

All the above applies also for environment variables `CFLAGS`, `CXXFLAGS`, etc.

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

closes #7

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin rerender

File: recipe/deactivate-gcc.sh [hunk 1]
```diff
@@ -117,11 +117,11 @@ _tc_activation \
...
-  "CPPFLAGS,${CPPFLAGS:-${CPPFLAGS_USED}}" \
-  "CFLAGS,${CFLAGS:-${CFLAGS_USED}}" \
-  "LDFLAGS,${LDFLAGS:-${LDFLAGS_USED}}" \
-  "DEBUG_CPPFLAGS,${DEBUG_CPPFLAGS:-${DEBUG_CPPFLAGS_USED}}" \
-  "DEBUG_CFLAGS,${DEBUG_CFLAGS:-${DEBUG_CFLAGS_USED}}" \
+  "CPPFLAGS,${CPPFLAGS_USED} ${CPPFLAGS:-}" \
+  "CFLAGS,${CFLAGS_USED} ${CFLAGS:-}" \
+  "LDFLAGS,${LDFLAGS_USED} ${LDFLAGS:-}" \
+  "DEBUG_CPPFLAGS,${DEBUG_CPPFLAGS_USED} ${DEBUG_CPPFLAGS:-}" \
+  "DEBUG_CFLAGS,${DEBUG_CFLAGS_USED} ${DEBUG_CFLAGS:-}" \
```
Comment (beckermr) [inline recipe/deactivate-gcc.sh:124]:
```suggestion
  "DEBUG_CFLAGS,${DEBUG_CFLAGS_USED} ${DEBUG_CFLAGS:-}" \
```

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
ok @isuruf this one is ready for review!

**Comment by isuruf** [comment]
cc @katietz

**Comment by isuruf** [comment]
Thanks

**Comment by xhochy** [comment]
Somehow this broke the tensorflow builds, when I pin `gcc_linux-64` to a previous build, everything works again. It is probably introducing somewhere an additional space or so. I'll investigate and update here.

**Comment by xhochy** [comment]
The issue is that we now have trailing whitespace or double whitespaces in these flags. I fixed this now via https://github.com/conda-forge/tensorflow-feedstock/pull/213/commits/701c1faa8a01674f283e9660cd18a16e98128fa3

**Comment by beckermr** [comment]
I'm happy to fix but idk a great bashism to prefix if a variable is not empty. I could use an if statement but that'd be horrible looking. Any ideas?

File: recipe/activate-gcc.sh [hunk 2]
```diff
@@ -158,11 +158,11 @@ _tc_activation \
   "CONDA_TOOLCHAIN_HOST,@CHOST@" \
   "CONDA_TOOLCHAIN_BUILD,@CBUILD@" \
   cc cpp gcc gcc-ar gcc-nm gcc-ranlib \
-  "CPPFLAGS,${CPPFLAGS:-${CPPFLAGS_USED}}" \
-  "CFLAGS,${CFLAGS:-${CFLAGS_USED}}" \
-  "LDFLAGS,${LDFLAGS:-${LDFLAGS_USED}}" \
-  "DEBUG_CPPFLAGS,${DEBUG_CPPFLAGS:-${DEBUG_CPPFLAGS_USED}}" \
-  "DEBUG_CFLAGS,${DEBUG_CFLAGS:-${DEBUG_CFLAGS_USED}}" \
+  "CPPFLAGS,${CPPFLAGS_USED} ${CPPFLAGS:-}" \
```
Comment (isuruf) [inline recipe/activate-gcc.sh:161]: Change to "CPPFLAGS,${CPPFLAGS_USED}"
and then add the following above line 156
```
[ -z "$CPPFLAGS" ] || CPPFLAGS_USED="${CPPFLAGS} ${CPPFLAGS_USED}"
```

Comment (beckermr) [reply recipe/activate-gcc.sh:161]: Thanks! I'll have to do all of them I think.

Comment (isuruf) [reply recipe/activate-gcc.sh:161]: You can also do,
```
CPPFLAGS,${CPPFLAGS_USED}${CPPFLAGS:+ }${CPPFLAGS:-}
```

Comment (beckermr) [reply recipe/activate-gcc.sh:161]: see https://github.com/conda-forge/ctng-compiler-activation-feedstock/pull/68

**Pull request merged**